src/metabase/lib/aggregation.cljc

```diff
@@ -360,19 +360,14 @@
                     (m/update-existing
                      :columns
                      (fn [cols]
-                       (let [refs (mapv lib.ref/ref cols)
-                             match (lib.equality/find-closest-matching-ref
-                                    (lib.options/update-options agg-col dissoc :temporal-unit)
-                                    refs)]
-                         (if match
-                           (mapv (fn [r c]
-                                   (cond-> c
-                                     (= r match) (assoc :selected? true)
-
-                                     (some? agg-temporal-unit)
-                                     (lib.temporal-bucket/with-temporal-bucket agg-temporal-unit)))
-                                 refs cols)
-                           cols)))))))
+                       (let [cols (lib.equality/mark-selected-columns
+                                   cols
+                                   [(lib.options/update-options agg-col dissoc :temporal-unit)])]
+                         (mapv (fn [c]
+                                 (cond-> c
+                                   (some? agg-temporal-unit)
+                                   (lib.temporal-bucket/with-temporal-bucket agg-temporal-unit)))
+                               cols)))))))
```

Comment:
Am I reading it right that this is removing the `:temporal-unit` then adding it back, in order to suit `mark-selected-columns`? That seems backwards.

If I'm misreading it, then a comment is in order because it's not clear why this juggling the temporal values is needed. (And if it is needed, why only temporal units and not bucketing or other extras?)

Reply:
I'm not 100% sure what it's doing TBH, I didn't write the original code, the only thing I changed here was updating it to use the new helper for marking columns selected instead of doing it manually